march_hardware/src/IMotionCube.cpp

```diff
@@ -506,18 +511,16 @@ bool IMotionCube::goToOperationEnabled()
               this->slaveIndex, angleRead, this->encoder.getLowerHardLimitIU(), this->encoder.getUpperHardLimitIU());
     throw std::domain_error("Joint outside hard limits");
   }
-  else
+
+  if (this->actuationMode == ActuationMode::position)
   {
-    if (this->actuationMode == ActuationMode::position)
-    {
-      this->actuateIU(angleRead);
-    }
-    if (this->actuationMode == ActuationMode::torque)
-    {
-      this->actuateTorque(0);
-    }
+    angleRead = this->getAngleIU();
```

Comment:
Why do this again?

Reply:
I guess you refer to the use in line 500 as well? Because the state of the encoder can change at any instant between the lines 500 and 517 it is best practice to read out the state again. This prevents unnecessary errors.

Reply:
Seems to me like overkill considering the program only has to calculate three if-statements in between 500 and 517 (because the code will error if any of the two above if statements is true). @Olavhaasie Opinion?

Reply:
I don't see how this would hurt our performance. And if we remove the extra check we run the slight risk of not sending the actual current angle to the joint. We'd be taking unnecessary risks, however slight, with no consequence to the functioning of our system.

Reply:
I think this is indeed uneccesary

Reply:
It is better not to put in a large number of unnecessary checks, it makes code longer than necessary and our descendants may think "no idea why this is here but it might be important so I don't dare remove it". 

Besides, I think that if you are talking about the risk of the encoder value changing, I think the bigger problem will appear if the encoder resets within that 0.1 nanoseconds, because then the motor will try to move to position 0.